### ISSUE TITLE
Database reorder

### ISSUE THREAD

**Issue by iulko** [duplicati/duplicati#5602]
**Is your feature request related to a problem? Please describe.**

I would like to have an easy way to reorder Backups. The way the order is important is when you want to have last Backup to be a backup of all your duplicati Databases. This way when you loose you AppData folder with all the backups you can do a disaster recovery of Database folder and restore Duplicati with full versions, settings and all the history in a easy and fast way etc

I already did it by decrypting server sqlite database on windows, changing ID of the backup to the highest one in Backup Table, than change from old ID to the new one in all other databases that mention BackupID and than if needed change sqlite_sequence also if we used a new higher ID.

**Describe alternatives you've considered**

A settings page in the frontend where you can change the order of the backups. If there is only reordering happening than we would not have to worry about sqlite_sequence, so maybe a simple button where you can swap ID, one for another and than the program would run the nessecary SQL queries. BackupID1 to BackupTemportaryID1000, BackupID2 to BackupID1, BackupTemportaryID1000 to BackupID2 would work.

**Comment by marceloduplicati** [comment]
This [PR](https://github.com/duplicati/duplicati/pull/6090) solves the UI/viewing part of it, but it does not actually change the IDs.

**Comment by kenkendk** [comment]
Closing this because we now have support for sorting the backups. It is not fully a drag-n-reorder type setup, but you can sort by various parameters including name, so you can set the names to follow the sort order you want.

**Comment by ts678** [comment]
> you can set the names to follow the sort order you want.

But job run order is unchanged and still based on ID, right?

**Comment by marceloduplicati** [comment]
@ts678 Yes, run job order is unchanged so far. 

What we've discussed is perhaps adding a new field to allow custom order without updating the ID and cascading the related records, but its been put to rest since one can now sort for viewing purposes on the frontend and API.

Do you believe there is an use case for actual ID re-ordering/re-assigning ?

**Comment by ts678** [comment]
> Do you believe there is an use case for actual ID re-ordering/re-assigning ?

I suspect this issue is it, although @iulko should clarify what issue intent is. I think it goes beyond appearance:

> I would like to have an easy way to reorder Backups. The way the order is important is when you want to have last Backup to be a backup of all your duplicati Databases.

The run-order wish makes complete sense, as you don't want to make a DR backup of job database that will be outdated afterwards.

How one gets that might depend on things like backup scheduling and queuing, which has changed somewhat just recently with PR:

[Run backups in order after startup #6394](https://github.com/duplicati/duplicati/pull/6394)

> Before, the order of backups being queued would depend on the database order, essentially creation order.

and that would be a motivation to do odd things like export/import, or edit the database to set IDs as desired.

The export/import also helps the visuals, but the new visual reordering now helps that. It doesn't help the runs.

For people who need very tight control of run order, sometimes ServerUtil or its 2.0 predecessor are suggested.
Putting those in a run-script-after or similar can force a chain to run as desired, but how to guarantee its finish?

If needed, I can probably go searching in Forum and Issues for people who care about run ordering. Cases exist.

**Comment by iulko** [comment]
> I suspect this issue is it, although [@iulko](https://github.com/iulko) should clarify what issue intent is. I think it goes beyond appearance:
> 

Yes, It goes beyond appearance. I want to set order of jobs running. But if there was a way to do it witchout changing the IDs, something like a new field "order" than that would be fine too.

> and that would be a motivation to do odd things like export/import, or edit the database to set IDs as desired.

yes, and I spend many hours restoring my database after fail attempts :<

> If needed, I can probably go searching in Forum and Issues for people who care about run ordering. Cases exist.

I saw atleast a dozen people asking for that

**Comment by ts678** [comment]
Some examples (which cite other examples):

[Backup order after “missed” time schedule](https://forum.duplicati.com/t/backup-order-after-missed-time-schedule/20626) was a good recent discussion of another must-run-last shutdown/unmount case.

[In what order does missed backup jobs run?](https://forum.duplicati.com/t/in-what-order-does-missed-backup-jobs-run/6329) is an old one without much authoritative information, wanting to control order.

[Reordering jobs](https://forum.duplicati.com/t/reordering-jobs/6633) wants a slow and less important job to run last. I speculate that the perceived top-down order is in ID order.

I haven't gone looking for the people who are backing up their job databases (false positive risk), but discussions happened.

**Comment by marceloduplicati** [comment]
Thank you @ts678 & @iulko, I will re-open the issue and, pun intended, get it sorted.

**Comment by ts678** [comment]
My first link above talks about FIFO queue order, which was expected by the developer, but not actually so until listed change.
I'm in there talking about what happens if desired sequence doesn't finish, for example if system turns off before it completes.
Above-cited "Run backups in order after startup #6394" cited issue where user had reason for earliest-schedule job to go first.

I'm still wondering if one can make all cases schedule as desired, especially given interruptions of varied length at varied spots.
Schedules can also differ in interval as well as time. Altogether it's lots of variables and moving parts to think about. Good luck.

Still, I "think" the increasingly FIFO nature in the upcoming release will be useful, and the missed-backups change might as well.
The former might allow people doing manual runs to try to force order by order they click "Run" in, or maybe scheduling order, expecting only one backup to run at once, therefore only minimal schedule difference is enough to queue up the backup order.

At least if behavior can be well defined and documented, then guidance will help users set up -- and explain if it doesn't always schedule as intended. Possibly improvements can be proposed, either by thinking through cases now, or by user feedback later.

@iulko has a specific case, and might have seen others. I'm wondering how many involve queuing, either by adding faster than completions (backup times can vary, especially if a compact happens), or by having the system down thus missing the schedule.

I'm not sure where else ID would come in. Backups started when system is up and idle run right then, regardless of any ordering.

**Comment by iulko** [comment]
I have all backups set to same time/every day but I manually run them by opening tray icon exe file when I want to run all backups and they than all start running automatically and the order they run is from database order. I think even if they were running on different hours they would still follow this order as time from last run is more than 24 hours.
